index.js

```diff
@@ -23,36 +24,27 @@ class CumulusMessageAdapterExecutionError extends Error {
  * @param {Object} input - the input to be sent to the message-adapter
  * @returns {Promise.<Object>} - the output of the message-adapter
  */
-function callCumulusMessageAdapter(command, input) {
-  return new Promise((resolve, reject) => {
-    const adapterDir = process.env.CUMULUS_MESSAGE_ADAPTER_DIR || './cumulus-message-adapter';
-    const cumulusMessageAdapter = cp.spawn('python', [adapterDir, command]);
-
-    // Collect STDOUT
-    let cumulusMessageAdapterStdout = '';
-    cumulusMessageAdapter.stdout.on('data', (chunk) => {
-      cumulusMessageAdapterStdout += chunk;
-    });
-
-    // Collect STDERR
-    let cumulusMessageAdapterStderr = '';
-    cumulusMessageAdapter.stderr.on('data', (chunk) => {
-      cumulusMessageAdapterStderr += chunk;
-    });
-
-    cumulusMessageAdapter.on('close', (code) => {
-      if (code === 0) resolve(JSON.parse(cumulusMessageAdapterStdout));
-      else reject(new CumulusMessageAdapterExecutionError(cumulusMessageAdapterStderr));
-    });
-
-    // If STDIN is closed already, something went wrong.  Don't handle it
-    // here and expect a non-zero code when the close event fires.
-    // If this line is not present, calling sled.stdin.end with a chunk will
-    // result in an unhandled "Error: write EPIPE".
+async function callCumulusMessageAdapter(command, input) {
+  const adapterDir = process.env.CUMULUS_MESSAGE_ADAPTER_DIR || './cumulus-message-adapter';
+  const systemPython = await lookpath('python');
+  let spawnArguments = [systemPython, [`${adapterDir}`, command]];
```

Comment:
This feels a little easier to read:

```js
let spawnArguments;
if (systemPython) {
  spawnArguments = [systemPython, [`${adapterDir}`, command]];
} else {
  spawnArguments = [`${adapterDir}/cma`, [command]];
}
```

or

```js
const spawnArguments = systemPython
  ? [systemPython, [`${adapterDir}`, command]]
  : [`${adapterDir}/cma`, [command]];
```

Reply:
The default with catch block idiom tends to result in lower logic complexity/indentation, generally,  .... but fair enough, at the expense of explicit readability, particularly in simple cases.       

bbc5c29